src/ORM/Connect/Database.php

```diff
@@ -343,12 +343,8 @@ public function manipulate($manipulation)
                         $query->addWhere(array('"ID"' => $writeInfo['id']));
                     }
 
-                    // Test to see if this update query shouldn't, in fact, be an insert
-                    if ($query->toSelect()->count()) {
```

Comment:
In any case #7073 has more details.

Imma close this PR as it adds little value keeping it open.